src/AspNet.Security.OAuth.Twitch/TwitchAuthenticationOptions.cs

```diff
@@ -25,10 +25,28 @@ public TwitchAuthenticationOptions()
             TokenEndpoint = TwitchAuthenticationDefaults.TokenEndpoint;
             UserInformationEndpoint = TwitchAuthenticationDefaults.UserInformationEndpoint;
 
-            Scope.Add("user_read");
+            Scope.Add("user:read:email");
+
+            ClaimActions.MapCustomJson(ClaimTypes.NameIdentifier, user =>
+            {
+                return user["data"]?[0]?["id"]?.ToObject<string>();
+            });
+
+            ClaimActions.MapCustomJson(ClaimTypes.Name, user =>
+            {
+                return user["data"]?[0]?["login"]?.ToObject<string>();
+            });
+
+            ClaimActions.MapCustomJson(ClaimTypes.GivenName, user =>
+            {
+                return user["data"]?[0]?["display_name"]?.ToObject<string>();
```

Comment:
Is the display name guaranteed to be the first/given name? If no, a different claim name should be used.

Reply:
We discussed this too but we can not find a claim which really fits. 
```json
{
    "data": [
        {
            "id": "1111",
            "login": "zaflu",
            "display_name": "zAfLu",
            "type": "",
            "broadcaster_type": "partner",
            "description": "",
            "profile_image_url": "LINKTOIMAGE",
            "offline_image_url": "LINKTOIMAE",
            "view_count": 000000,
            "email": "xxxx@yyyy.yy"
        }
    ]
}
```
So the display_name shows the real write shape. 

What claim would you choose?

Reply:
Feel free to use a custom claim name. E.g https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/blob/dev/src/AspNet.Security.OAuth.Bitbucket/BitbucketAuthenticationConstants.cs#L16